src/lib.rs

```diff
@@ -123,6 +131,9 @@ lazy_static! {
             PlottingBackend::None
         }
     };
+}
+#[cfg(feature = "cargo_criterion_support")]
+lazy_static! {
     static ref CARGO_CRITERION_CONNECTION: Option<Mutex<Connection>> = {
         match std::env::var("CARGO_CRITERION_PORT") {
```

Comment:
Currently, when `cargo_criterion_support` is not enabled, `criterion` behaves as if `cargo-criterion` does not exist. It might be useful to check somewhere whether the `CARGO_CRITERION_PORT` env variable is set, and print a warning or error if `cargo_criterion_support` is not enabled. This wouldn't be the right place; it would need to go in the benchmark's `main` function (similar to the previous warnings when `cargo_bench_support` was not enabled).